e2e/test/app.ts

```diff
@@ -0,0 +1,194 @@
+import * as assert from "assert";
+import * as CodePush from "rest-definitions";
+import * as Q from "q";
+import { Command } from "./utils/command";
+import { Error } from "./utils/error";
+import { makeRandomString } from "./utils/misc";
+import { Success } from "./utils/success";
+import { Validate } from "./utils/validate";
+var nixt = require("nixt");
+var tryJSON = require("try-json");
+
+export function appTests() {
+    
+    // To keep the account clean of excessive apps, we must remember what apps we create and destroy them after the tests!
+    
+    var appNamesAddedDuringTests: string[] = [];
+    
+    function addAndDone(done: MochaDone, appNameAdded: string): (error: any) => void {
+        return (error: any) => {
+            appNamesAddedDuringTests.push(appNameAdded);
+            done(error);
+        }
+    }
+    
+    function removeAndDone(done: MochaDone, appNameRemoved: string): (error: any) => void {
+        return (error: any) => {
+            appNamesAddedDuringTests.splice(appNamesAddedDuringTests.indexOf(appNameRemoved), 1);
+            done(error);
+        }
+    }
```

Comment:
remove `addAndDone` and `removeAndDone`
Use the frameworks functionality to add and cleanup the data that you need. Mocha provides `beforeEach` and `afterEach` hooks. Use them.

```
 beforeEach(function() {
    // runs before each test in this block
  });

  afterEach(function() {
    // runs after each test in this block
  });
```